asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py

```diff
@@ -409,25 +410,30 @@ def create_fec_dataset(
              The planned FEC network which can be executed locally or submitted to alchemiscale.
         """
         # check that all ligands are unique in the series
-        if len(set(ligands)) != len(ligands):
-            count = Counter(ligands)
-            duplicated = [
-                key.compound_name for key, value in count.items() if value > 1
-            ]
-            raise ValueError(
-                f"ligand series contains {len(duplicated)} duplicate ligands: {duplicated}"
-            )
+        if ligands:
```

Comment:
We should unpack the ligands from the graphml input here as well and make sure the names are unique

Reply:
Moved this to the new `from_graphml` constructor.